src/main-win32.h

```diff
@@ -0,0 +1,28 @@
+#ifndef MAIN_WIN32_H
+#define MAIN_WIN32_H
+
+#ifdef WIN32
+#include <stringapiset.h>
+#include <wchar.h>
+
+#define DEFINE_MAIN(ARGC, ARGV) \
```

Comment:
Haven't notice this code before. Is this used when building native for windows, not via emulation layers like msys2 or cygwin?

Reply:
I think, when you compile with `-municode` on windows, `wmain()` is called instead of `main()` as starting point with `wchar_t*` strings instead of `char*` strings; then that code in `wmain()` converts the `wchar_t*` strings to `char*` strings encoding with utf-8.

Reply:
I assume it is probably an hack to make the program get the correct utf-8 encoding on unicode characters you pass through `cmd.exe` because windows encodings are weird.
For example, recently, I added a test to `curl` that verifies that utf-8 bytes get correctly encoded to JSON; and I had to modify my test to get the utf-8 string from a file instead of from command line arguemnts because the windows CI was not passing the right bytes to `curl`. :/
https://github.com/curl/curl/pull/12434